src/components/Calculator.js

```diff
@@ -1,46 +1,68 @@
 import React from 'react';
 import './claculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      display: '0',
+
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange = (event) => {
+    const id = event.target.textContent.trim();
+    this.setState((prevState) => ({ ...calculate(prevState, id) }), this.displayOnScreen);
+  }
+
+  displayOnScreen() {
+    const { total, next } = this.state;
+    if (!next && !total) {
+      this.setState({ display: '0' });
+    } else {
+      this.setState({ display: next ?? total });
+    }
   }
 
   render() {
+    const { display } = this.state;
     return (
       <div className="claculator">
         <div className="result">
-          0
+          {display}
         </div>
         <div className="panel">
-          <button type="submit" className="btn"> AC </button>
-          <button type="submit" className="btn"> +/- </button>
-          <button type="submit" className="btn"> % </button>
-          <button type="submit" className="operation"> / </button>
+          <button onClick={this.handleChange} type="submit" id="allClear" className="btn"> AC </button>
+          <button onClick={this.handleChange} type="submit" id="switchSign" className="btn"> +/- </button>
+          <button onClick={this.handleChange} type="submit" id="percent" className="btn "> % </button>
+          <button onClick={this.handleChange} type="submit" id="division" className="operation"> / </button>
```

Comment:
- [ ] The UI disappears when I try to divide a number using the `/` operator, kindly fix it.